compiler/rustc_resolve/src/late.rs

```diff
@@ -4572,7 +4632,8 @@ impl<'ast> Visitor<'ast> for LifetimeCountVisitor<'_, '_, '_> {
             | ItemKind::ExternCrate(..)
             | ItemKind::MacroDef(..)
             | ItemKind::GlobalAsm(..)
-            | ItemKind::MacCall(..) => {}
+            | ItemKind::MacCall(..)
+            | ItemKind::Delegation(..) => {}
```

Comment:
This is a bit subtle.
Delegated functions have lifetimes, their count is not necessarily zero.
But skipping the delegation items here doesn't mean that the count will be considered zero, it means there will be a panic when retrieving the count, but for delegation items we are never actually retrieving that count in practice.
This case needs a comment.